torch/optim/adadelta.py

```diff
@@ -286,12 +286,12 @@ def _multi_tensor_adadelta(
 
     grouped_tensors = Optimizer._group_tensors_by_device_and_dtype([params, grads, square_avgs, acc_deltas])
     for ((device_params, device_grads, device_square_avgs, device_acc_deltas), _) in grouped_tensors.values():
-        if maximize:
```

Comment:
Why are these these changes necessary?

It turns out we weren't testing complex with maximize before, and so we ran into failures when I turned on testing.
The crux is that foreach ops return tuples, which are not mutable, but the _`view_as_real` requires mutable lists. So when maximize came before, that turned `grouped_grads` into a tuple which was not good for `_view_as_real`.